File: src/main/java/istu/bacs/submission/LanguageConverter.java
package istu.bacs.submission;

import org.springframework.stereotype.Component;

import javax.persistence.AttributeConverter;

@Component
public class LanguageConverter implements AttributeConverter<Language, Integer> {

    @Override
    public Integer convertToDatabaseColumn(Language language) {
        return language.getLanguageId();
    }

    @Override
    public Language convertToEntityAttribute(Integer languageId) {
        return Language.findById(languageId);
    }
}

File: src/main/java/istu/bacs/submission/LanguageConverter.kt
package istu.bacs.submission

import org.springframework.stereotype.Component
import javax.persistence.AttributeConverter

@Component
class LanguageConverter : AttributeConverter<Language, Int> {
    override fun convertToDatabaseColumn(language: Language?) = language!!.languageId
    override fun convertToEntityAttribute(id: Int?) = Language.findById(id!!)
}